rspace/src/main/scala/coop/rchain/rspace/package.scala

```diff
@@ -141,17 +141,20 @@ package object rspace {
                   |at <channels: $channels>""".stripMargin.replace('\n', ' '))
             None
           case Some(dataCandidates) =>
-            dataCandidates
-              .sortBy(_.datumIndex)(Ordering[Int].reverse)
-              .foreach {
-                case DataCandidate(candidateChannel, Datum(_, persistData), dataIndex)
-                    if !persistData =>
-                  store.removeDatum(txn, Seq(candidateChannel), dataIndex)
-                case _ =>
-                  ()
-              }
-            logger.debug(s"consume: data found for <patterns: $patterns> at <channels: $channels>")
-            Some((continuation, dataCandidates.map(_.datum.a)))
+            store.eventsCounter.registerConsumeCommEvent {
```

Comment:
It appears that you are measuring the duration of this block.  This isn't the right strategy - a COMM Event is not what happens inside of this block.  Rather it's the event of satisfying a match (or matches in the case of a join) above.  So the "event" from our perspective is when the call to `extractDataCandidates` has returned a `Some(...)`, so my comment before was to suggest that you increment the counter here.

In terms of a time-based metric, what we really need to be able to measure is how many times we are hitting this counter per second.  We also need to do this for produces and consumes.